pkg/providers/tinkerbell/config/template-md.yaml

```diff
@@ -86,6 +94,35 @@ spec:
             anonymous-auth: "false"
 {{- if .kubeletExtraArgs }}
 {{ .kubeletExtraArgs.ToYaml | indent 12 }}
+{{- end }}
+{{- if and .registryMirrorConfiguration (ne .format "bottlerocket") }}
+      files:
+{{- end }}
```

Comment:
I think this `end` is in the wrong place. It should wrap everything to be specified under `files` else there's a risk a susbequent condition could evaluate as true and populate invalid YAML (a YAML missing `files:`).